### PR TITLE
[Product Block Editor]: add Linked product sections. First approach.

### LINKED ISSUE #42915: When the user selects Linked products, we show two sections, Upsells and Cross-sells.
**Prerequisites (mark completed items with an [x]):**
- [ ] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [ ] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**ACs**

```[tasklist]
### Tasks
- [x] Create a new `Linked products` tab. AkVNGImLgSqCObTQ3idVn7-fi-12744_189608
- [x] Create a new empty `Upsells` section. AkVNGImLgSqCObTQ3idVn7-fi-12744_189608
- [x] Create a new empty `Cross-sells` section. AkVNGImLgSqCObTQ3idVn7-fi-12744_189608
- [x] Make sure the whole tab is managed by the `product-linked` feature flag made in https://github.com/woocommerce/woocommerce/issues/42914
- [x] When the user clicks on the Question Mark icon next to either of the section titles, we show the relevant tooltip with help text and a link to learn more about linked products.
- [x] When the product is a Grouped Product or an External/Affiliate Product, the Cross-sell section should be hidden. Only the Upsell section should appear.
```

### PR DESCRIPTION
**Pull request by retrofox**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Depens on https://github.com/woocommerce/woocommerce/pull/43009
Part of https://github.com/woocommerce/woocommerce/issues/42915
Closes https://github.com/woocommerce/woocommerce/issues/42915

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure the `product-linked` flag is enabled 
2. Go to the Product Editor
3. Select the `Linked Products` section
4. Confirm you see the `Upsell` and `Cross-sells` section
5. Confirm the text and link of the help tooltip

Upsells | Cross-sells
-----|------
image | image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

**Review by nathanss — CHANGES_REQUESTED** [review]
The code looks good, you just need to validate scenarios where the group is null (feature flag disabled). Otherwise, it will cause errors for everyone who has the feature flag disabled.

**Review by nathanss — APPROVED** [review]

File: plugins/woocommerce/src/Internal/Features/ProductBlockEditor/ProductTemplates/SimpleProductTemplate.php [hunk 1]
```diff
@@ -1104,4 +1105,50 @@ private function add_variation_group_blocks() {
 			)
 		);
 	}
+
+	/**
+	 * Adds the linked products group blocks to the template.
+	 */
+	private function add_linked_products_group_blocks() {
+		if ( ! Features::is_enabled( 'product-virtual-downloadable' ) ) {
```
Comment (mdperez86) [inline plugins/woocommerce/src/Internal/Features/ProductBlockEditor/ProductTemplates/SimpleProductTemplate.php:1113]: @retrofox @nathanss This is not the right feature flag to validate. Also I didn't see the problem commented there https://github.com/woocommerce/woocommerce/pull/43009/files#r1433178112 already solved here.

Comment (retrofox) [reply plugins/woocommerce/src/Internal/Features/ProductBlockEditor/ProductTemplates/SimpleProductTemplate.php:1113]: > @retrofox @nathanss This is not the right feature flag to validate.

🤦 Creating a new PR

> Also I didn't see the problem commented there https://github.com/woocommerce/woocommerce/pull/43009/files#r1433178112 already solved here.

As previously mentioned in the same thread, the issue will be addressed in a follow-up.
After all, isn't the purpose of hiding the implementation behind a feature flag?

Do you see something wrong that could impact the users? I don’t know, something like showing part of this implementation even when the feature flag is desirable?

Comment (retrofox) [reply plugins/woocommerce/src/Internal/Features/ProductBlockEditor/ProductTemplates/SimpleProductTemplate.php:1113]: https://github.com/woocommerce/woocommerce/pull/43119

**Pull request merged**